### PR TITLE
Initial version or XML wrapper with libxml2 support.

### LINKED ISSUE #17: Allow clients to add different XML parsers
We should allow clients to provide and use different XML parsers (platform provided XML parsers?). This may require a slim wrapper for the XML/DOM function calls in  SVG Native Viewer.

### PR DESCRIPTION
**Pull request by dirkschulze**
Fixes #17 and adds 2 XML parser as example:
* libxml2
* RapidXML (boost)

### CODE REVIEW

**Comment by dirkschulze** [comment]
This patch does various optimizations and fixes to compensate the bigger lib size for the XML wrapper. The performance was improved too. The wrapper has no negative impact on performance.

**Comment by dirkschulze** [comment]
@qrb7xc It would be great if you could take a look as well. If there is anything I can do to make the wrapper thinner or more efficient, please let me know.

File: svgnative/CMakeLists.txt [hunk 1]
```diff
@@ -113,6 +115,8 @@ else()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pedantic -Wextra")
 endif()
 
+#set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
```
Comment (dirkschulze) [inline svgnative/CMakeLists.txt]: will remove

Comment (mpsuzuki) [reply svgnative/CMakeLists.txt]: good!

**Pull request merged**